src/tests.rs

```diff
@@ -68,9 +68,16 @@ macro_rules! testtrip {
 
             #[test]
             fn roundtrip_frame() {
-                use super::{frame_depress, frame_press};
+                use super::{read_frame_depress, write_frame_press};
                 let d = &$data[..];
-                assert_eq!(d, &*frame_depress(&frame_press(d)));
+                assert_eq!(d, &*read_frame_depress(&write_frame_press(d)));
+            }
+
+            #[test]
+            fn read_and_write_frame_encoder_match() {
+                use super::{read_frame_press, write_frame_press};
+                let d = &$data[..];
+                assert_eq!(read_frame_press(d), write_frame_press(d));
```

Comment:
@BurntSushi This is the most important of the new tests. It's part of your `testtrip!` macro, and for every round-trip test case you already have, it verifies that `read::FrameEncoder` and `write::FrameEncoder` produce identical output. Does this seem sufficient to you?